lib/rules/color-no-out-gamut-range/index.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const validateOptions = require('../../utils/validateOptions');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+// @ts-expect-error -- TS7016
+const Color = require('colorjs.io').default;
+const report = require('../../utils/report');
+const getDeclarationValue = require('../../utils/getDeclarationValue');
+const { isAtRule } = require('../../utils/typeGuards');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+
+const ruleName = 'color-no-out-gamut-range';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (color) => `Unexpected color out of gamut range "${color}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/color-no-out-gamut-range',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			// ignore css variable declaration
+			if (decl.prop.startsWith('--')) return;
+
+			// ignore scss variable declaration
+			if (decl.prop.startsWith('$')) return;
```

Comment:
[suggest] It's nice to consider non-standard syntaxes! 👍🏼
In this case, the `isStandardSyntaxProperty` utility is helpful. For example:

```suggestion
            if (!isStandardSyntaxProperty(decl.prop)) return;

			const parsedValue = valueParser(getDeclarationValue(decl));
```

Reply:
In addition, the `decl.prop.startsWith('--')` code can be replaced with the `isCustomProperty` utility.

Reply:
> [suggest] It's nice to consider non-standard syntaxes! 👍🏼 In this case, the `isStandardSyntaxProperty` utility is helpful. For example:

I guess, we can improve this rule in the future. It' very inspiring to contribute to open source tools like this, so maybe I will even implement it myself :)